src/plugins/controls/public/control_group/control_group_renderer.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import uuid from 'uuid';
+import useLifecycles from 'react-use/lib/useLifecycles';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+
+import { IEmbeddable } from '@kbn/embeddable-plugin/public';
+
+import { pluginServices } from '../services';
+import { getDefaultControlGroupInput } from '../../common';
+import { ControlGroupInput, ControlGroupOutput, CONTROL_GROUP_TYPE } from './types';
+import { ControlGroupContainer } from './embeddable/control_group_container';
+
+export interface ControlGroupRendererProps {
+  input?: Partial<Pick<ControlGroupInput, 'viewMode' | 'executionContext'>>;
+  onEmbeddableLoad: (controlGroupContainer: ControlGroupContainer) => void;
+}
+
+export const ControlGroupRenderer = ({ input, onEmbeddableLoad }: ControlGroupRendererProps) => {
```

Comment:
Just to be clear, this `ControlGroupRenderer` isn't actually **used** anywhere yet, correct? You are simply introducing it to use it in a follow-up PR related to portable dashboards.

Reply:
Absolutely correct.